tekton/release-pipeline.yaml

```diff
@@ -0,0 +1,123 @@
+---
+apiVersion: tekton.dev/v1alpha1
+kind: Pipeline
+metadata:
+  name: pipeline-release
+spec:
+  params:
+  - name: package
+    description: package to release
+    default: github.com/tektoncd/pipeline
+  - name: imageRegistry
+    default: gcr.io/tekton-releases # REPLACE with your own registry
+  - name: versionTag
+    description: The X.Y.Z version that the artifacts should be tagged with
+  resources:
+  - name: source-repo
+    type: git
+  - name: bucket
+    type: storage
+  - name: builtBaseImage
+    type: image
+  - name: builtEntrypointImage
+    type: image
+  - name: builtKubeconfigWriterImage
+    type: image
+  - name: builtCredsInitImage
+    type: image
+  - name: builtGitInitImage
+    type: image
+  - name: builtNopImage
+    type: image
+  - name: builtBashImage
+    type: image
+  - name: builtGsutilImage
+    type: image
+  - name: builtControllerImage
+    type: image
+  - name: builtWebhookImage
+    type: image
+  - name: builtDigestExporterImage
+    type: image
+  - name: builtPullRequestInitImage
+    type: image
+  tasks:
+    - name: lint
+      taskRef:
+        name: golangci-lint
+      params:
+        - name: package
+          value: ${params.package}
+        - name: flags
+          value: -v
+      resources:
+        inputs:
+          - name: source
+            resource: source-repo
+    - name: unit-tests
+      runAfter: [lint]
+      taskRef:
+        name: golang-test
+      params:
+        - name: package
+          value: ${params.package}
+      resources:
+        inputs:
+          - name: source
+            resource: source-repo
+    - name: build
+      runAfter: [lint]
+      taskRef:
+        name: golang-build
+      params:
+        - name: package
+          value: ${params.package}
+      resources:
+        inputs:
+          - name: source
+            resource: source-repo
+    - name: publish-images
+      runAfter: [build, unit-tests]
```

Comment:
I think we could add lint here to the list, if any of the three fails we do not publish images